docs/content/docs/dev/table/types.md

```diff
@@ -1487,6 +1481,88 @@ Not supported.
 {{< /tab >}}
 {{< /tabs >}}
 
+Casting
+-------
+
+Flink SQL can perform casting between a defined pair `from` type and `target` type. While some
+casting operations can always succeed no matter the input value, others can fail at runtime, that
+is, for some input values there is no way to create a value for the target type. For example, it's
+always possible to convert `INT` to `STRING`, but viceversa is not true, as you can provide an
+input `STRING` which cannot be converted to `INT`, like `'abc'`.
```

Comment:
```suggestion
Flink SQL can perform casting between a defined `from` type and `target` type pair. While some
casting operations can always succeed regardless of the input value, others can fail at runtime 
(i.e. when there is no way to create a value for the target type). For example, it is always possible
to convert `INT` to `STRING`, but you can provide an input `STRING` which cannot be converted 
to `INT`.
```

Reply:
I modified a bit the last sentence, check it out